plutus-use-cases/src/Plutus/Contracts/Uniswap.hs

```diff
@@ -911,3 +958,27 @@ userEndpoints us =
         tell $ Last $ Just $ case e of
             Left err -> Left err
             Right () -> Right Stopped
+
+
+
+
+PlutusTx.makeIsDataIndexed ''U [('U, 0)]
+PlutusTx.makeLift ''U
```

Comment:
I'm surprised you need these! I would have hoped that we were clever enough to not require the constraints on all the type parameters if they're not needed for the actual instances...

Reply:
This is the error if they're left out, if you're interested:

```
src/Plutus/Contracts/Uniswap/Data.hs:94:1: error:
    • No instance for (PlutusTx.IsData U)
        arising from a use of ‘PlutusTx.toData’
    • In the expression: PlutusTx.toData arg_augI
      In the second argument of ‘PlutusTx.Constr’, namely
        ‘[PlutusTx.toData arg_augI]’
      In the expression: (PlutusTx.Constr 0) [PlutusTx.toData arg_augI]
   |
94 | PlutusTx.makeIsDataIndexed ''Uniswap [('Uniswap, 0)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Plutus/Contracts/Uniswap/Data.hs:94:1: error:
    • No instance for (PlutusTx.IsData U)
        arising from a use of ‘PlutusTx.fromData’
    • In the second argument of ‘(<*>)’, namely
        ‘PlutusTx.fromData arg_augJ’
      In the expression: (pure Uniswap <*> PlutusTx.fromData arg_augJ)
      In an equation for ‘PlutusTx.fromData’:
          PlutusTx.fromData (PlutusTx.Constr i_augK [arg_augJ])
            | (i_augK == 0) = (pure Uniswap <*> PlutusTx.fromData arg_augJ)
   |
94 | PlutusTx.makeIsDataIndexed ''Uniswap [('Uniswap, 0)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Plutus/Contracts/Uniswap/Data.hs:95:1: error:
    • No instance for (PlutusTx.Typeable
                         plutus-core-0.1.0.0:PlutusCore.Universe.Default.DefaultUni U)
        arising from a use of ‘PlutusTx.Lift.Class.typeRep’
    • In the first argument of ‘PlutusTx.Lift.Class.CompileType’, namely
        ‘(PlutusTx.Lift.Class.typeRep
            (Data.Proxy.Proxy :: Data.Proxy.Proxy U))’
      In the first argument of ‘PlutusTx.Lift.Class.unCompileType’, namely
        ‘(PlutusTx.Lift.Class.CompileType
            (PlutusTx.Lift.Class.typeRep
               (Data.Proxy.Proxy :: Data.Proxy.Proxy U)))’
      In the second argument of ‘(Prelude.$)’, namely
        ‘PlutusTx.Lift.Class.unCompileType
           (PlutusTx.Lift.Class.CompileType
              (PlutusTx.Lift.Class.typeRep
                 (Data.Proxy.Proxy :: Data.Proxy.Proxy U)))’
   |
95 | PlutusTx.makeLift ''Uniswap
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
Suspicious. Now the question is whether this bothers me enough to go and debug the awful TH code :D

Reply:
:scream_cat:

Reply:
I did some digging and sadly it's not trivial: https://jira.iohk.io/browse/SCP-2212